be/src/pipeline/exec/scan_operator.cpp

```diff
@@ -979,18 +979,19 @@ Status ScanLocalState<Derived>::_prepare_scanners() {
             scanner->set_query_statistics(_query_statistics.get());
         }
         COUNTER_SET(_num_scanners, static_cast<int64_t>(scanners.size()));
-        RETURN_IF_ERROR(_start_scanners(_scanners));
+        RETURN_IF_ERROR(start_scanners(_scanners));
     }
     return Status::OK();
 }
 
 template <typename Derived>
-Status ScanLocalState<Derived>::_start_scanners(
+Status ScanLocalState<Derived>::start_scanners(
```

Comment:
The only difference between this and `file_scan_operator` is that `file_scan_operator` pass the `query_parallel_instance_num` separately.
So I suggest to create a new virtual method `get_query_parallel_instance_num()`,
and for `scan_operator`, it returns `state()->query_parallel_instance_num()`.
For `file_scan_operator`, it returns `_batch_split_mode ? 1 : state()->query_parallel_instance_num()`

Reply:
If extract this function to scan operator, It seems that the runtime state cannot be obtained through state() except through the parameters of this function. But this feels a bit strange.

Reply:
Done.